### ISSUE TITLE
Tower type inventory update fails with ansible@devel

### ISSUE THREAD

**Issue by kdelee** [ansible/awx#4580]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

Trying to run tower-type inventory sync (like when you use a different tower as an inventory source) fails with ansible@devel.

NOTE: ansible 2.9 feature freeze is August 29 AFAIK.

##### ENVIRONMENT
* AWX version: devel
* Ansible version:  devel

##### STEPS TO REPRODUCE

Create a tower inventory source, attempt to run inventory update

##### EXPECTED RESULTS

Works

##### ACTUAL RESULTS

```
Using /etc/ansible/ansible.cfg as config file  [WARNING]:  * Failed to parse /tmp/awx_2789_hg1ql6mm/tower.yml with auto plugin: Invalid type for configuration option plugin_type: inventory plugin: tower setting: inventory_id : Invalid type provided for "string": 1513    File "/usr/lib/python2.7/site-packages/ansible/inventory/manager.py", line 280, in parse_source     plugin.parse(self._inventory, self._loader, source, cache=cache)   File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/auto.py", line 58, in parse     plugin.parse(inventory, loader, path, cache=cache)   File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/tower.py", line 150, in parse     self._read_config_data(path)   File "/usr/lib/python2.7/site-packages/ansible/plugins/inventory/__init__.py", line 229, in _read_config_data     self.set_options(direct=config)   File "/usr/lib/python2.7/site-packages/ansible/plugins/__init__.py", line 75, in set_options     self._options = C.config.get_plugin_options(get_plugin_class(self), self._load_name, keys=task_keys, variables=var_options, direct=direct)   File "/usr/lib/python2.7/site-packages/ansible/config/manager.py", line 348, in get_plugin_options     options[option] = self.get_config_value(option, plugin_type=plugin_type, plugin_name=name, keys=keys, variables=variables, direct=direct)   File "/usr/lib/python2.7/site-packages/ansible/config/manager.py", line 409, in get_config_value     keys=keys, variables=variables, direct=direct)   File "/usr/lib/python2.7/site-packages/ansible/config/manager.py", line 502, in get_config_value_and_origin     (to_native(_get_entry(plugin_type, plugin_name, config)), to_native(e)))   [WARNING]: Unable to parse /tmp/awx_2789_hg1ql6mm/tower.yml as an inventory source  ERROR! No inventory was parsed, please check your configuration and options.  TIME WHEN STATUS WAS FOUND: 2019-08-23 14:45:34.907173 (UTC)
```

Key portion of this error:
```
Invalid type for configuration option plugin_type: inventory plugin: tower setting: inventory_id : Invalid type provided for "string": 1513    File "/usr/lib/python2.7/site-packages/ansible/inventory/manager.py", line 280, in parse_source     plugin.parse(self._inventory, self._loader, source, cache=cache)   
```
##### ADDITIONAL INFORMATION

@AlanCoding you mentioned https://github.com/ansible/ansible/issues/59442

**Comment by AlanCoding** [comment]
tl;dr on the linked openstack issue, before, the plugin config parsing did not validate type, and now it does.

https://github.com/ansible/awx/blob/6e6676adb3cfece83313b0a3717abd952e733552/awx/main/models/inventory.py#L2687-L2688

https://github.com/ansible/ansible/blob/bc4ef99533514f0d35c24ddfe0fd4497ceb1ac8c/lib/ansible/plugins/inventory/tower.py#L49-L54

@kdelee Those changes to the config parsing have been merged since July 17th, would it be relatively plausible that this has been behaving this way since that time?

**Comment by kdelee** [comment]
@AlanCoding I have been seeing problems on YOLO for a while,  I think we were just slow to figure out what was going on because we have not been running the full test suite on devel on all platforms (which is unfortunate and I thought we were doing that :cry: )

**Comment by AlanCoding** [comment]
Hoping it is resolved in https://github.com/ansible/ansible/issues/61333

**Comment by AlanCoding** [comment]
With the general rubric of fix suggested here:

https://github.com/ansible/ansible/pull/61335#issuecomment-524998367

I feel like we are confident enough that this can and should be handled specific to Ansible 2.9, and there is no need to track in AWX.
